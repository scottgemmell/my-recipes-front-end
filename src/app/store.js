import { configureStore } from '@reduxjs/toolkit';
import counterReducer from '../features/counter/counterSlice';
import recipesReducer from '../features/recipes/recipesSlice';

export default configureStore({
  reducer: {
    counter: counterReducer,
	recipes: recipesReducer,
  },
});
