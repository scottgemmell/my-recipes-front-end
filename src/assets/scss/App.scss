@import "./_mixins";
@import "./_site-header";

%u-visually-hidden,
.u-visually-hidden {
	display: none;
}

%u-spinner,
.u-spinner {
	// height: 100vh;
	// width: 100vw;
	display: flex;
	justify-content: center;
	align-items: center;
}

* {
	box-sizing: border-box;
}

body {
	margin: 0;
	font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Roboto', 'Oxygen',
		'Ubuntu', 'Cantarell', 'Fira Sans', 'Droid Sans', 'Helvetica Neue',
	sans-serif;
	-webkit-font-smoothing: antialiased;
	-moz-osx-font-smoothing: grayscale;
}

code {
	font-family: source-code-pro, Menlo, Monaco, Consolas, 'Courier New',
	monospace;
}

.top {
	background-color: #fafafa;
}

.layer {
	.layer__inner {
		max-width: 960px;
		margin: 0 auto;
		//background-color: gold;
		padding: 0 20px;
	}
}



h1 {
	border-bottom: 2px solid #f4f4f4;
	padding-bottom: 20px;
}


/* .app-link {
  color: #61dafb;
  text-decoration: none;
} */

.thumb {
	position: relative;
	// height: 210px;
	// width: 280px;
	//margin-bottom: 40px;
	list-style-type: none;
	display: block;
	margin: 10px;

	&:hover {
		.thumb__meta {
			background-color: white;
		}
	}

	// :hover {
		
	// 	.thumb__meta:before {
	// 		content: "";
	// 		color: green;
	// 		position: absolute;
	// 		top: 0;
	// 		right: 0;
	// 		bottom: 0;
	// 		left: 0;
	// 		background-color: rgba(255,255,255,1);
	// 	}
	// }

	.thumb__display {
		position: relative;
		img {
			max-width: 100%;
			line-height: 1;
			// position: absolute;
  			// clip: rect(0px,300px,400px,0px);
		}
	}

	.thumb__meta {
		background-color: rgba(255,255,255,0.6);
		position: absolute;
		z-index: 10;
		bottom: 0;
		left: 0;
		padding: 5px 10px 10px 10px;

		

		h3 {
			margin-top: 0;
			margin-bottom: 0;

			svg {
				position: relative;
				top: 3px;
			}

			a {
				color: #000;
				text-decoration: none;
				
				
			}
		}

		p {
			margin: 0 0 10px 0;
			display: none;
			// @include respond-to('medium') {
			// 	display: block;
			// }
		}
	}
}

// .search-results {
// 	margin: 0;
// 	padding: 0;
// 	display: grid;
// 	grid-template-columns: 1fr 1fr 1fr;
// 	grid-gap: 20px;

// 	li {
// 		list-style-type: none;
// 		margin: 0;
// 		padding: 0;
// 	}
// }

.subtitle {
	display: flex;
	align-items: center;

	svg {
		margin-right: 4px;
	}
}


.recipes-list {
	display: grid;
	grid-template-columns: 1fr;
	margin: 0 -10px;
	padding: 0;

	@include respond-to('small') {
		grid-template-columns: 1fr 1fr;
	}
	
	@include respond-to('medium') {
		grid-template-columns: 1fr 1fr 1fr;
	}
	
	@include respond-to('large') {
		grid-template-columns: 1fr 1fr 1fr 1fr;
	}
}

main {
	text-align: center;

	@include respond-to('small') {
		text-align: left;
	}
}

.ingredients-list {
	margin: 0 -10px 40px -10px;
	padding: 0;

	// @include respond-to('small') {
	// 	grid-template-columns: 1fr;
	// }
	
	// @include respond-to('medium') {
	// 	grid-template-columns: 1fr 1fr;
	// }
	
	// @include respond-to('large') {
	// 	grid-template-columns: 1fr 1fr;
	// }
}

.ingredient {
	position: relative;
	list-style-type: none;
	display: block;
	margin: 10px;
	
	.ingredient__inner {
		display: flex;
		flex-direction: row-reverse;
		justify-content: flex-end;
		align-items: center;
		background-color:#f0f0f0;
	}

	.ingredient__meta {
		
		padding: 40px;
		flex-grow: 1;

		h3 {
			margin-top: 0;
			margin-bottom: 0;

			svg {
				position: relative;
				top: 3px;
			}

			a {
				color: #000;
				text-decoration: none;
			}
		}

	}
}

.form {
	width: 400px;
}

.field {
	margin-bottom: 20px;

	.field__title {
		font-weight: bold;
		margin-bottom: 5px;
	}

	.field__controls {
		input,
		textarea {
			width: 100%;
			padding: 10px;
		}

		textarea {
			min-height: 150px;
		}
	}

	.field__optional {
		font-weight: normal;
		color: grey;
		float: right;
	}
}

.controls {
	display: flex;
	justify-content: flex-end;
}

.button {
	padding: 5px 10px;
}

.site-theme {
	float: right;
	margin-top: 0;
	margin-bottom: 0;
	padding-top: 10px;
}