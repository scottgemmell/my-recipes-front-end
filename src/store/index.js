import { configureStore } from "@reduxjs/toolkit";
import { recipesApi } from "../api/recipesApi";

export default configureStore({
  reducer: {
	[recipesApi.reducerPath]: recipesApi.reducer,
  },
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware().concat(recipesApi.middleware),
});
