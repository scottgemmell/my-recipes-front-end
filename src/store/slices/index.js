import { recipesSlice } from "./recipesSlice";

const rootReducer = {
	recipes: recipesSlice.reducer,
}

export default rootReducer;