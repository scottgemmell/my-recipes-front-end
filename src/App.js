import React from "react";
import { BrowserRouter as Router, Route, Redirect, Switch } from "react-router-dom";
import { 
	PageAdd, 
	PageEdit, 
	PageHome, 
	PageLogin, 
	PageProfile, 
	PageRegister,
	PageView, 
} from "./pages/";
import { Main, SiteFooter, SiteHeader } from "./common";
import "./assets/scss/App.scss";

//export const RecipesContext = React.createContext([]);

function App() {

	return (
		<div className="App">
			<Router>
				<SiteHeader />
				<Main>
					<Switch>
						<Route 
							exact={true}
							path={["/", "/list/"]} 
							component={PageHome} 
						/>
						<Route 
							exact={true}
							path={"/profile/"} 
							component={PageProfile}
						/>
						<Route 
							exact={true}
							path={"/add/"} 
							component={PageAdd}
						/>
						
						<Route 
							exact={true}
							path={"/register/"} 
							component={PageRegister}
						/>
						<Route 
							exact={true}
							path={"/login/"} 
							component={PageLogin}
						/>
						<Route 
							path={`/view/:id/:recipe/`} 
							component={PageView}
						/>
						<Route 
							path={`/edit/:id/:recipe/`} 
							component={PageEdit}
						/> 
						<Redirect to="/login/" />
					</Switch>
				</Main>
				<SiteFooter />
			</Router>
		</div>
	);
}

export default App;
