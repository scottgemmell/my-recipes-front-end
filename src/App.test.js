import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import ReactDOM from "react-dom";
import store from './store';
import App from './App';

describe("Describe", () => {
	test('renders learn react link', () => {
		const { getByText } = render(
		  <Provider store={store}>
			<App />
		  </Provider>
		);
		expect(getByText("MY RECIPES")).toBeInTheDocument();
	  });
	  
	  test('renders without crashing', () => {
		  const app = document.createElement('div');
		  expect(app).toMatchSnapshot() /* This is our snapshot */
		  ReactDOM.unmountComponentAtNode(app);
	  });	  
});

