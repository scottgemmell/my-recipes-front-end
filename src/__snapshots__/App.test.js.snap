// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Describe renders without crashing 1`] = `<div />`;
